master/internal/checkpoint_gc.go

```diff
@@ -47,10 +48,15 @@ func (t *checkpointGCTask) Receive(ctx *actor.Context) error {
 			return errors.Wrap(err, "cannot start a new task session for a GC task")
 		}
 
-		config := t.experiment.Config.CheckpointStorage
+		config := t.experiment.Config.CheckpointStorage()
 
-		checkpoints, err := t.db.ExperimentCheckpointsToGCRaw(t.experiment.ID,
-			&config.SaveExperimentBest, &config.SaveTrialBest, &config.SaveTrialLatest, true)
+		checkpoints, err := t.db.ExperimentCheckpointsToGCRaw(
+			t.experiment.ID,
+			ptrs.IntPtr(config.SaveExperimentBest()),
```

Comment:
It could be, but semantically `ptrs.IntPtr(config.SaveExperimentBest())` is more correct because `config.RawSaveExperimentBest` could semantically return `nil` wheras it cannot be `nil` as written.

Also, using the Raw API here violates the rule of "never use the Raw API after calling WithDefaults", even though technically read-only usage of the Raw API is always safe.  I thought once about having WithDefaults return separate structs so that you really physically couldn't use the Raw API after WithDefaults, but when I tried to implement that it was much too complex.

Also, I was very confused why this call takes a pointer at all.